src/coreclr/vm/i386/cgenx86.cpp

```diff
@@ -200,6 +200,41 @@ void EHContext::UpdateFrame(PREGDISPLAY regs)
 }
 #endif // FEATURE_EH_FUNCLETS
 
+
+void UnmanagedToManagedFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
+{
+    CONTRACT_VOID
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        HOST_NOCALLS;
+        SUPPORTS_DAC;
+    }
+    CONTRACT_END;
+
+    pRD->PCTAddr = GetReturnAddressPtr();
```

Comment:
This seems odd. The input is marked `const` but we are updating it?

Reply:
In this case the const-ness is on the pointer itself, not on the value it points to.  I'm not sure why the const is on the parameter of the function (it is an override of a base virtual one that has it that way), maybe it is called on a const Frame derived object somewhere.
The PREGDISPLAY is a typedef `typedef REGDISPLAY *PREGDISPLAY` and that causes the `const PREGDISPLAY` to apply the const-ness  to the pointer. In other words, it is as if the argument was `REGDISPLAY * const pRD`.

Reply:
@janvorli I get the C++ semantics here, that isn't my concern. It is the intent. My guess here is the original author didn't realize the `typedef` nuance here so I am simply trying to reconcile what the `const` should mean rather than what it semantically does in the language. Should we change it?

Reply:
To make sure I understand, you'd prefer the `const` to be removed to not to confuse people into thinking the contents of the struct is const?

Reply:
Ideally, yes. However, it was merely an observation. Once I checked saw the `typedef` I knew exactly what was going on, but it is one of those odd things I've notice people can trip up on. If the `const` isn't really doing anything, I think it would be good to remove. That is simply quality of life stuff though and if it is more than one or two places I'd leave it.

Reply:
It is at 54 places in 12 files

Reply:
> It is at 54 places in 12 files

Definitely not worth it.